### ISSUE TITLE
🐛 HTML parser doesn't parse `<? ?>` syntax

### ISSUE THREAD

**Issue by ematipico** [biomejs/biome#10612]
### Environment information

main

### What happened?

The following

```html
<?xml version="1.0" encoding="utf-8"?>
```
```
public/img/favicon.svg:1:1 parse 

  ✖ Unescaped `<` bracket character. Expected a tag or escaped character.

  > 1 │ <?xml version="1.0" encoding="utf-8"?>
      │ ^
    2 │ <svg width="64" height="64" version="1.0" xmlns="http://www.w3.org/2000/svg">
    3 │  <rect id="background" width="100%" height="100%" fill="none" />

  ℹ Replace this character with `&lt;` to escape it.
```

### Expected result

It shouldn't throw an error

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by lgarron** [comment]
I'm also running into this with certain SVGs exported from vector programs. I can ignore them for now, but it would certainly be nice not to crash by default when they are in a repo.

**Comment by flying-sheep** [comment]
Yeah, definitely need to support these for SVGs, Inkscape generates them.
